src/enterprise/extension.ts

```diff
@@ -132,20 +132,31 @@ function registerAuthenticationProviders(
   );
 }
 
-async function uninstallGATabnineIfPresent() {
-  // search for the GA extension
-  const tabnine = extensions.getExtension("tabnine.tabnine-vscode");
-  if (tabnine) {
-    // in this case we want to uninstall the GA tabnine extension
+async function uninstallAllOtherExtensionsIfPresent() {
+  return uninstallOtherTabnineIfPresent([
+    "tabnine.tabnine-vscode",
+    "tabnine.tabnine-vscode-enterprise",
+  ]);
+}
+
+async function uninstallOtherTabnineIfPresent(extensionIds: string[]) {
+  const oldExtensions = extensionIds.map((extensionId) =>
+    extensions.getExtension(extensionId)
+  );
+  if (oldExtensions && oldExtensions.length) {
     const uninstall = await confirm(
       "⚠️ You have a conflicting version of Tabnine!",
       "Fix"
     );
     // the user provided consent
     if (uninstall) {
-      await commands.executeCommand(
-        "workbench.extensions.uninstallExtension",
-        "tabnine.tabnine-vscode"
+      await Promise.all(
```

Comment:
if it fails it will retry again next time? when is this triggered?
also, promiseAll rejects on first error. is this what you need?

Reply:
it will happen again on the next startup, yes.
I don't care if it will be rejected, as it will try again 🤷🏻